style/js/README.md

```diff
@@ -1,3 +1,190 @@
+JavaScript Style Guides
+=====
+
+Guides for programing JavaScript in style
+
+General
+---
+* Use [JSHint](http://jshint.com/)
+* Always use 'use strict'
+
+Parens and Braces
+---
+* Leave one space between 'if' and '('.
+
+```javascript
+if (something) {
+  //some work here;
+}
+```
+
+* Prefer this form for multiline if statments
+
+```javascript
+if (something) {
+  // do stuff
+  // even more stuff
+} else {
+  // some other stuff
+  // real stuff
+}
+```
+
+* On if statments, whenever it makes sense, you can omit the '{}'
+
+```javascript
+if (something)
+  return stuff;
+else
+  return otherStuff;
+```
+
+* Don't include spaces between 'function' and '('.
+
+```javascript
+function bar() {
+  return  foo;
+}
+
+var bar = function() {
+  return foo;
+}
+```
+
+Assignments and Declarations
+---
+* Initialize arrays with '[ ]'
+
+```javascript
+var array = [];
+```
+
+* Initialize objects with '{ }'
+
+```javascript
+var object = {};
+```
+
+* Leave one space after '{' and before '}' when initializing objects on the same line
+
+```javascript
+var map = { ready: 9, when: 4, "you are": 15 };
+```
+
+* Prefer this form for multiline object/array initialization
+
+```javascript
+var map = {
+  when: 4,
+  ready: 9,
+  "you are": 15
+};
+
+var map = [
```

Comment:
not really. I often do this for short lines as well, when I want to add some visual structure. For instance, I just looked at my other monitor and noticed that I've written this:

```javascript
var frontVertices = [
  0,     1 * rowHeight,
  width, 3 * rowHeight,
  width, 4 * rowHeight
];
```

In this example, each 2 values represents a point, and that visual organization helps a lot. And I do this a lot in other cases as well. I guess it depends a lot more on context rather than just the length of the line

PS: and please don't bother commenting on how `rowHeight` is duplicated, or each point should have their own object. That's beside the point, and is dependent on the API I'm using.